dvc/repo/experiments/executor/base.py

```diff
@@ -457,25 +457,21 @@ def _repro_args(cls, dvc):
 
     @staticmethod
     def _auto_push(git_remote: str, dvc: "Repo", scm: "Git"):
-        from dvc.repo.experiments.push import push
-
         if git_remote == scm.root_dir:
             logger.warning(
```

Comment:
one other comment on this is that we will end up repeating this warning for every commit, it would be better if we just checked for this condition once at the start of `reproduce()`